File: Schule/DA-Abgabe/prototype/backend/src/main/java/com/siemens/backend/api/graphqlTypes/modelObjects/Container.java
package com.siemens.backend.api.graphqlTypes.modelObjects;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor

public class Container {
    UUID guid;
    String name;
    String className;
    Float nominalVoltage;
    List<SubContainer> subContainer;
}

File: Schule/DA-Abgabe/prototype/frontend/src/app/_models/container.ts
import { ClassNameInfo } from './modelObject';

export interface ContainerInfo {
  guid: string;
  name?: string;
  nominalVoltage?: number;
  className?: ClassNameInfo;
}

export interface Container {
  guid: string;
  name: string;
  className: string;
  nominalVoltage?: number;
  subcontainer: Container[];
}
